### PR TITLE
Fix for the Drawer Overlay issue

### LINKED ISSUE #31724: [Bug]: Drawer Overlay from bottom gets off the canvas/window in mobile browsers on opened
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 11th Gen Intel(R) Core(TM) i5-11260H @ 2.60GHz
    Memory: 11.29 GB / 23.73 GB
  Browsers:
    Edge: Chromium (125.0.2535.92)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/devbox/epic-flower-ndtcmn

## Actual Behavior
When opening the drawer in **an Android mobile browser (Edge, Chrome, Firefox)**, it opens from the bottom and grows to the top of viewport, but then it countinues to grwo until getting the top part of the drawer off canvas
image

## Expected Behavior
In **an Android mobile browser (Chrome, Edge, Firefox)** When opening the drawer from the bottom with full size, you expect to cover exactly the full page
image

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by tummala-manasa**
## Previous Behavior
When opening the drawer in an Android mobile browser (Edge, Chrome, Firefox), it opens from the bottom and grows to the top of viewport, but then it continues to grow until getting the top part of the drawer off canvas

## New Behavior
The browser task bar does not cover the content.

## Related Issue(s)

- Fixes #31724

### CODE REVIEW

**Comment by tummala-manasa** [comment]
> ```
> @microsoft-github-policy-service agree
> ```

@microsoft-github-policy-service agree

**Review by marcosmoura — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.097 MB`<br />`270.957 kB` | `1.097 MB`<br />`270.959 kB` | `1 B` <br />`2 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218 kB`<br />`63.151 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/48c6239e3b34bf42c56ee76159ab5cb43f8b34da'>48c6239e3b34bf42c56ee76159ab5cb43f8b34da</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">35</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Avatar-mount_base.html">629</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Avatar-mount.html">678</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Button-mount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Button-mount.html">328</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Field-mount_base.html">1116</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Field-mount.html">1139</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProvider-mount_base.html">724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProvider-mount.html">717</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-mount_base.html">87</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-mount.html">92</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">35</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/MakeStyles-mount_base.html">857</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/MakeStyles-mount.html">860</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Persona-mount_base.html">1807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/Persona-mount.html">1758</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/SpinButton-mount_base.html">1409</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/SpinButton-mount.html">1376</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/SwatchPicker-mount_base.html">1703</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32060/perf-test-react-components/SwatchPicker-mount.html">1670</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-drawer-6332ed54-ea15-403c-a872-673eee9ec783.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-drawer-6332ed54-ea15-403c-a872-673eee9ec783.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32060&commitId=7920a92d71ea73722a9b90de67d240fc249a2a2a&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|16| Changed|
|Avatar Converged.badgeMask.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**